lib/internal/errors.js

```diff
@@ -622,6 +622,45 @@ function addNumericalSeparator(val) {
   return `${val.slice(0, i)}${res}`;
 }
 
+// Used to enhance the stack that will be picked up by the inspector
+const kEnhanceStackBeforeInspector = Symbol('kEnhanceStackBeforeInspector');
+
+// These are supposed to be called only on fatal exceptions before
+// the process exits.
+const fatalExceptionStackEnhancers = {
+  beforeInspector(error) {
+    if (typeof error[kEnhanceStackBeforeInspector] !== 'function') {
+      return error.stack;
+    }
+
+    try {
+      // Set the error.stack here so it gets picked up by the
+      // inspector.
+      error.stack = error[kEnhanceStackBeforeInspector]();
+    } catch {
+      // We are just enhancing the error. If it fails, ignore it.
+    }
+    return error.stack;
+  },
+  afterInspector(error) {
+    const originalStack = error.stack;
+    const {
+      inspect,
+      inspectDefaultOptions: {
+        colors: defaultColors
+      }
+    } = lazyInternalUtilInspect();
+    const colors = internalBinding('util').guessHandleType(2) === 'TTY' &&
+                   require('internal/tty').hasColors() ||
+                   defaultColors;
```

Comment:
On the windows issue: we could temporarily disable the coloring by adding `process.plaform === 'win32'` to the conditions here, though I would prefer an alternative fix with changing the implementation of `PrintErrorString()` to use `uv_write` for TTY instead. I have a WIP in https://github.com/joyeecheung/node/tree/print-uv-write which still fails a few tests that I need to investigate. Either way that's kind of orthognal to this PR as it only tries to fix https://github.com/nodejs/node/issues/28287